### PR TITLE
Replace oasis-deposit with Oasis CLI

### LINKED ISSUE #617: Replace oasis-deposit with Oasis CLI
For depositing initial ROSE to accounts, we are currently using a custom go app `oasis-deposit`. Since the Oasis CLI is now part of the docker image use it instead. Try to keep `-test-mnemonic`, `-n` and `-to` parameters behavior the same, if feasible.

Depends on #613. Depends on https://github.com/oasisprotocol/cli/issues/213

### PR DESCRIPTION
**Pull request by abukosek**
Requires Oasis CLI 0.10.2+

Closes #617.

### CODE REVIEW

**Comment by abukosek** [comment]
image

Merging https://github.com/oasisprotocol/cli/pull/304 and a new release of Oasis CLI is required for this to work in the emerald-localnet image as well.

File: docker/common/start.sh [hunk 1]
```diff
@@ -218,18 +256,90 @@ else
...
 if [ ! -z "${OASIS_CLI_BINARY:-}" ]; then
   ${OASIS_CLI_BINARY} network add-local localnet unix:/serverdir/node/net-runner/network/client-0/internal.sock -y
   ${OASIS_CLI_BINARY} paratime add localnet ${PARATIME_NAME} 8000000000000000000000000000000000000000000000000000000000000000 --num-decimals 18 -y
   ${OASIS_CLI_BINARY} network set-default localnet
 fi
 
+
+notice "Populating accounts...\n\n"
+
+# Remove underscores from amount, as those are only there to make it easier
```
Comment (matevz) [inline docker/common/start.sh]: Can you move the new code inside a function, e.g. `populate_accounts()`

**Comment by matevz** [comment]
Also feel free to remove `docker/common/oasis-deposit` folder.

**Comment by matevz** [comment]
[Oasis CLI 0.10.2](https://github.com/oasisprotocol/cli/releases/tag/v0.10.2) was released. Feel free to try it out.

File: docker/README.md [hunk 2]
```diff
@@ -6,7 +6,6 @@ set up localnet for developing dApps running on Sapphire or Emerald:
 - oasis-node and oasis-net-runner
 - Sapphire or Emerald ParaTime .orc bundles
 - oasis-web3-gateway
-- oasis-deposit helper to fund local development accounts
```
Comment (matevz) [inline docker/README.md:11]:
```suggestion
- the Oasis CLI for funding initial accounts
- the Oasis Nexus indexer and Explorer frontend

```

File: docker/common/start.sh [hunk 3]
```diff
@@ -118,6 +117,119 @@ else
...
+    ACCT_OUT=""
+    PK_OUT=""
+    for n in $(seq 0 $((N-1)))
+    do
+      acct="account$n"
+
+      # Parse Ethereum address and private key from output.
+      IFS="#"
+      out=$(${OASIS_CLI_BINARY} wallet export ${acct} -y)
+      eth_addr=$(echo "${out}" | head -5 | tail -1 | cut -d':' -f2 | awk '{$1=$1};1')
```
Comment (matevz) [inline docker/common/start.sh]: Please grep for `Ethereum address: ` instead of relying on line number.

File: docker/common/start.sh [hunk 4]
```diff
@@ -118,6 +117,119 @@ else
...
+      pk=$(echo "${out}" | tail -1)
```
Comment (matevz) [inline docker/common/start.sh:195]: Same here.

File: docker/common/start.sh [hunk 5]
```diff
@@ -118,6 +117,119 @@ else
...
+    done
+
+    ACCT_OUT=""
+    PK_OUT=""
+    for n in $(seq 0 $((N-1)))
+    do
+      acct="account$n"
+
+      # Parse Ethereum address and private key from output.
+      IFS="#"
```
Comment (matevz) [inline docker/common/start.sh:192]: Leftover?

**Review by matevz — APPROVED** [review]
Some nits. Otherwise looks good.

File: docker/common/start.sh (hunk 4 above)
Comment (abukosek) [reply docker/common/start.sh:195]: The private key is always on the last line and there is no other text around it to grep for, so this is the nicest solution.

File: docker/common/start.sh (hunk 5 above)
Comment (abukosek) [reply docker/common/start.sh:192]: No, this is here by design :)

Hint: Add `echo "${out}"` and try it with and without this line :)

**Pull request merged**